modules/node_groups/node_groups.tf

```diff
@@ -14,8 +14,8 @@ resource "aws_eks_node_group" "workers" {
   }
 
   ami_type        = lookup(each.value, "ami_type", null)
-  disk_size       = lookup(each.value, "disk_size", null)
-  instance_types  = each.value["launch_template_id"] != null ? [] : [each.value["instance_type"]]
+  disk_size       = each.value["launch_template_id"] != null || each.value["create_launch_template"] ? null : lookup(each.value, "disk_size", null)
+  instance_types  = each.value["launch_template_id"] != null || each.value["create_launch_template"] ? [] : [each.value["instance_type"]]
```

Comment:
It'd be cool to add `capacity_type   = lookup(each.value, "capacity_type", "ON_DEMAND")` right above this, so users could choose to use `"SPOT"`

Reply:
If think this is implemented by https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1129